src/main/java/net/fabricmc/loader/api/LanguageAdapter.java

```diff
@@ -16,6 +16,87 @@
 
 package net.fabricmc.loader.api;
 
+/**
+ * Creates instances of objects from custom notations.
+ *
+ * <p>It enables obtaining of other JVM languages' objects with custom instantiation logic.</p>
+ *
+ * <p>A language adapter is defined as so in {@code fabric.mod.json}:
+ * <pre><blockquote>
+ *   "languageAdapter": {
+ *     "&lt;a key&gt;": "&lt;the binary name of the language adapter class&gt;"
+ *   }
+ * </blockquote></pre>
+ * Multiple keys can be present in the {@code languageAdapter} section.</p>
+ *
+ * <p>In the declaration, the language adapter is referred by its <a href="https://docs.oracle.com/javase/specs/jls/se8/html/jls-13.html#jls-13.1">binary name</a>,
+ * such as {@code "mypackage.MyClass$Inner"}. It must have a no-argument public constructor for the Loader to instantiate.</p>
```

Comment:
This isnt quite right I dont think, as its based on the implimentation

Reply:
🤔 well, this is the technical name of the "dot format"

Reply:
no, the 2nd bit about the constructor

Reply:
Hmm, no? https://github.com/FabricMC/fabric-loader/blob/9f6b85907d239a3e40e1476140a69998cf06f7ca/src/main/java/net/fabricmc/loader/FabricLoader.java#L347
Here loader is explicitly requiring calling the constructor